polaris-core/src/main/java/org/apache/polaris/core/auth/AuthenticatedPolarisPrincipal.java

```diff
@@ -48,14 +48,23 @@ public PolarisEntity getPrincipalEntity() {
     return principalEntity;
   }
 
+  /**
+   * Returns the set of activated principal role names. Activated role names are the roles that were
+   * explicitly requested by the client when authenticating, through JWT claims or other means.
+   *
+   * <p>By convention, this method returns an empty set when the principal is requesting all
+   * available principal roles.
+   */
```

Comment:
I'm trying to document things as I go. This is the best knowledge I can get for this principal attribute.